pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/ScanBasedFilterOperator.java

```diff
@@ -73,32 +77,17 @@ public String getOperatorName() {
     return OPERATOR_NAME;
   }
 
-  public static class ScanBlock extends BaseFilterBlock {
-
-    private FilterBlockDocIdSet docIdSet;
-
-    public ScanBlock(FilterBlockDocIdSet docIdSet) {
-      this.docIdSet = docIdSet;
-    }
-
-    @Override
-    public BlockValSet getBlockValueSet() {
-      throw new UnsupportedOperationException("getBlockValueSet not supported in " + this.getClass());
-    }
-
-    @Override
-    public BlockDocIdValueSet getBlockDocIdValueSet() {
-      throw new UnsupportedOperationException("getBlockDocIdValueSet not supported in " + this.getClass());
-    }
-
-    @Override
-    public BlockMetadata getMetadata() {
-      throw new UnsupportedOperationException("getMetadata not supported in " + this.getClass());
-    }
+  /**
+   * Returns the predicate evaluator associated with the scan filter.
+   */
+  public PredicateEvaluator getPredicateEvaluator() {
+    return _predicateEvaluator;
+  }
 
-    @Override
-    public FilterBlockDocIdSet getFilteredBlockDocIdSet() {
-      return docIdSet;
-    }
+  /**
+   * Returns the metadata of the data source associated with the scan filter.
+   */
+  public DataSourceMetadata getDataSourceMetadata() {
```

Comment:
Just FYI. Kishore had a comment on my earlier PR that this should be changed to ColumnMetadata
https://github.com/linkedin/pinot/pull/3111#discussion_r213012688

Reply:
I noticed that, but it's a little bit hard to access ColumnMetadata in this layer. I just kept the current API but only for ScanBasedFilterOperator.